File: kattis/rationalsequence3.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int testcase;
	cin >> testcase;
	for (int i=0; i < testcase; i++) {
		int sets;
		cin >> sets;
		long long n;
		cin >> n;
		vector<long long> track;
		track.push_back(n);
		while (n > 1) {
			if (n%2) {
				n = (n-1)/2;
				track.push_back(n);
			}
			else {
				n /= 2;
				track.push_back(n);
			}
		}
		reverse(track.begin(), track.end());
		pair<long long, long long> final = make_pair(1, 1);
		for (int j = 1; j < track.size(); j++) {
			if (track[j-1] * 2 == track[j]) {
				long long num = final.first;
				long long denom = final.second;
				denom += num;
				final = make_pair(num, denom);
			}
			else {
				long long num = final.first;
				long long denom = final.second;
				num += denom;
				final = make_pair(num, denom);
			}
		}
		cout << sets << ' ' << final.first << '/' << final.second << endl;
	}
}

File: kattis/rationalsequence3.java
import java.util.*;

class Pair {
	public long first;
	public long second;

	public Pair(long first, long second) {
		this.first = first;
		this.second = second;
	}
}

class Main {
	public static void main(String[] args) {
		Kattio io = new Kattio(System.in);

		int tc = io.getInt();

		while (tc-- > 0) {
			ArrayList<Integer> track = new ArrayList<>();
			int k = io.getInt();
			int n = io.getInt();

			while (n > 1) {
				if (n % 2 == 1) {
					n = (n - 1) / 2;
					track.add(n);
				} else {
					n = n / 2;
					track.add(n);
				}
			}

			Pair p = new Pair(1, 1);
			for (int i = track.size() - 1; i > 0; i--) {
				if (track.get(i) == track.get(i - 1) * 2) {
					p.second += p.first;
				} else {
					p.first += p.second;
				}
			}

			io.println(k + " " + p.first + "/" + p.second);
		}

		io.flush();
	}
}